models/i18n.cfc

```diff
@@ -1,970 +1,1094 @@
-
-			// set locale setup on configuration file
-			setFWLocale( instance.defaultLocale, true );
-
-			// test for rb file and if it exists load it as the default resource bundle
-			if( len( instance.defaultResourceBundle ) ){
-				variables.resourceService.loadBundle( rbFile=instance.defaultResourceBundle, rbLocale=instance.defaultLocale, rbAlias="default" );
-			}
+/**
+ * Copyright Since 2005 ColdBox Framework by Luis Majano and Ortus Solutions, Corp
+ * www.ortussolutions.com
+ * ---
+ * Internationalization and localization support for ColdBox
+ */
+
+component singleton accessors="true" {
+
+	// DI
+	property name="resourceService" inject="resourceService@cbi18n";
+	property name="controller"      inject="coldbox";
+
+	// properties
+	property name="LocaleStorage";
+	property name="DefaultLocale";
+	property name="DefaultResourceBundle";
+
+	/**
+	 * Constructor
+	 */
+	function init() {
+		// Internal Java Objects
+		variables.aDateFormat = createObject( "java", "java.text.DateFormat" );
+		variables.aLocale     = createObject( "java", "java.util.Locale" );
+		variables.timeZone    = createObject( "java", "java.util.TimeZone" );
+		variables.aCalendar   = createObject( "java", "java.util.GregorianCalendar" ).init( buildLocale() );
+
+		// internal settings
+		variables.localeStorage         = "";
+		variables.defaultResourceBundle = "";
+		variables.defaultLocale         = "";
+
+		return this;
+	}
+
+	/**
+	 * Reads,parses,saves the locale and resource bundles defined in the config. Called only internally by the framework. Use at your own risk
+	 */
+	void function configure() {
+		// Default instance settings
+		variables.localeStorage         = variables.controller.getSetting( "LocaleStorage" );
+		variables.defaultResourceBundle = variables.controller.getSetting( "DefaultResourceBundle" );
+		variables.defaultLocale         = variables.controller.getSetting( "DefaultLocale" );
+
+		// set locale setup on configuration file
+		setFWLocale( variables.defaultLocale, true );
+
+		// test for rb file and if it exists load it as the default resource bundle
+		if ( variables.defaultResourceBundle.len() ) {
+			variables.resourceService.loadBundle(
+				rbFile   = variables.defaultResourceBundle,
+				rbLocale = variables.defaultLocale,
+				rbAlias  = "default"
+			);
+		}
 
-			// are we loading multiple resource bundles? If so, load up their default locale
-			var resourceBundles = variables.controller.getSetting( name="resourceBundles", defaultValue=structNew() );
-			if( structCount( resourceBundles ) ){
-				for( var thisBundle in resourceBundles ){
-					variables.resourceService.loadBundle( rbFile=resourceBundles[ thisBundle ], rbLocale=instance.defaultLocale, rbAlias=lcase( thisBundle ) );
-				}
+		// are we loading multiple resource bundles? If so, load up their default locale
+		var resourceBundles = variables.controller.getSetting( name = "resourceBundles", defaultValue = {} );
+		if ( resourceBundles.count() ) {
```

Comment:
you can remove the count(), each won't iterate if empty.  Less code.

Reply:
Why didn't I see this :-)
Done